### PR TITLE
Support Android LiteMode

### PR DESCRIPTION
**Pull request by rops**
Following discussion https://github.com/lelandrichardson/react-native-maps/issues/300

### CODE REVIEW

**Comment by spikebrehm** [comment]
Thanks, this looks good!

Can you squash the commits first?  There's 18 of them, with a lot of merge commits.

cc: @felipecsl to look at Android code.

File: android/src/main/java/com/airbnb/android/react/maps/AirMapManager.java [hunk 1]
```diff
@@ -44,8 +44,11 @@
 
     private final ReactApplicationContext appContext;
 
+    protected GoogleMapOptions googleMapOptions;
```
Comment (felipecsl) [inline android/src/main/java/com/airbnb/android/react/maps/AirMapManager.java]: final

File: android/src/main/java/com/airbnb/android/react/maps/AirMapLiteManager.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package com.airbnb.android.react.maps;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.google.android.gms.maps.GoogleMapOptions;
+
+public class AirMapLiteManager extends AirMapManager {
+
+    public static final String REACT_CLASS = "AIRMapLite";
```
Comment (felipecsl) [inline android/src/main/java/com/airbnb/android/react/maps/AirMapLiteManager.java]: private

**Comment by felipecsl** [comment]
android code lgtm after the 2 comments are addressed

**Comment by jrichardlai** [comment]
Should we add an example for `liteMode`?

**Comment by spikebrehm** [comment]
> Should we add an example for `liteMode`?

Great idea! Let's add that before merging.

**Comment by rops** [comment]
Adding Examples now

File: android/src/main/java/com/airbnb/android/react/maps/AirMapManager.java (hunk 1 above)
Comment (rops) [inline android/src/main/java/com/airbnb/android/react/maps/AirMapManager.java]: can't be final as AirMapLiteManager overrides it

**Comment by rops** [comment]
Added docs and examples

**Comment by rops** [comment]
Can someone take another quick look? I guess it's kind of ready to get in?

**Comment by jrichardlai** [comment]
tried the example, looks good :) 👍

**Pull request merged**